ui/src/app/utils/auth.tsx

```diff
@@ -0,0 +1,30 @@
+// utilities around Authority and page-based authorization
+
+import {Authority, Profile} from 'generated/fetch';
+
+// Admin pages guarded by a particular Authority
+enum AuthGuardedPage {
+    USER_ADMIN,
+    USER_AUDIT,
+    WORKSPACE_ADMIN,
+    WORKSPACE_AUDIT,
+    SERVICE_BANNER,
+    INSTITUTION_ADMIN,
+}
+
+const authorityByPage: Map<AuthGuardedPage, Authority> = new Map([
+    [AuthGuardedPage.USER_ADMIN, Authority.ACCESSCONTROLADMIN],
+    [AuthGuardedPage.USER_AUDIT, Authority.ACCESSCONTROLADMIN],
+    [AuthGuardedPage.WORKSPACE_ADMIN, Authority.RESEARCHERDATAVIEW],
+    [AuthGuardedPage.WORKSPACE_AUDIT, Authority.RESEARCHERDATAVIEW],
+    [AuthGuardedPage.SERVICE_BANNER, Authority.COMMUNICATIONSADMIN],
+    [AuthGuardedPage.INSTITUTION_ADMIN, Authority.INSTITUTIONADMIN],
+]);
+
+const hasAuthorityForPage = (profile: Profile, page: AuthGuardedPage): boolean =>
+    profile.authorities.includes(authorityByPage.get(page));
```

Comment:
Thank you for centralizing this. Could you also add the following logic while we're here to address this long-standing UI bug? https://precisionmedicineinitiative.atlassian.net/browse/RW-4852

```
profile.authorities.includes(Authority.DEVELOPER) || ...
```